integration-test/1810-alternate-viewpoints.py

```diff
@@ -264,3 +264,130 @@ def test_whole_claim(self):
 
         self.assertTrue(saw_xa, "Expected to see XA's viewpoint boundary")
         self.assertTrue(saw_xb, "Expected to see XB's country boundary")
+
+
+class PlaceTest(FixtureTest):
+
+    def test_disputed_capital(self):
+        import dsl
+
+        z, x, y = 16, 0, 0
+
+        # a country capital which CN doesn't think is a country capital. this
+        # is just a test case, and any resemblance to real disputes, living or
+        # dead, is purely coincidental.
+        self.generate_fixtures(
+            dsl.way(1, dsl.tile_centre_shape(z, x, y), {
+                'name': 'Foo',
+                'featurecla': 'Admin-0 capital',
+                'fclass_cn': 'Admin-1 capital',
+                'scalerank': 4,
+                'min_zoom': 4,
+                'source': 'naturalearthdata.com',
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'places', {
+                'kind': 'locality',
+                'country_capital': type(True),
+                'country_capital:cn': type(False),
+                'region_capital:cn': type(True),
+            })
+
+    def _check(self, default, dispute, expected):
+        import dsl
+
+        z, x, y = 16, 0, 0
+
+        self.generate_fixtures(
+            dsl.way(1, dsl.tile_centre_shape(z, x, y), {
+                'name': 'Foo',
+                'featurecla': default,
+                'fclass_iso': dispute,
+                'scalerank': 4,
+                'min_zoom': 4,
+                'source': 'naturalearthdata.com',
+            }),
+        )
+
+        expected['kind'] = 'locality'
+        self.assert_has_feature(z, x, y, 'places', expected)
```

Comment:
I don't understand this test, can you walk me thru it?

Reply:
Ah, I understand more now. Might be worth a comment above 298 `def _check` explaining the pattern since all the later tests use it.

Reply:
Good idea, added in 4c43b1c2.